### ISSUE TITLE
SugarFolder doing Insert instead of Update

### ISSUE THREAD

**Issue by tsmgeek** [salesagility/SuiteCRM#8718]
https://github.com/salesagility/SuiteCRM/blob/4ee24c78ebfa6d2c4ad6ec6e9bc8b9e590f7858d/include/SugarFolders/SugarFolders.php#L1226

When looking at this code it seems to me as if the logic a not right, its trying to do an INSERT even when an `id` is supplied.

Any ideas? else ile submit a PR.

**Comment by tsmgeek** [comment]
https://github.com/salesagility/SuiteCRM/pull/8720

This is the patch, if any issues please let me know.

**Comment by chris001** [comment]
@tsmgeek Excellent find.  In case you missed it, Travis ran the tests on PR #8720 , and they failed on this commit:
https://github.com/salesagility/SuiteCRM/pull/8720/commits/55017b2625dd7cf56815cc84a1e1b30dc3d69763

**Comment by tsmgeek** [comment]
@chris001  Ive just looked at the UnitTest on this and its failing correctly.

https://github.com/salesagility/SuiteCRM/blob/4ee24c78ebfa6d2c4ad6ec6e9bc8b9e590f7858d/tests/unit/phpunit/include/SugarFolders/SugarFolderTest.php#L709

Why would we `new_with_id` on a record that was already saved? Of course that will fail due to duplicate keys.

Shall I just change the unittest to make it correct?

**Comment by chris001** [comment]
> Why would we `new_with_id` on a record that was already saved? Of course that will fail due to duplicate keys.
> 
> Shall I just change the unittest to make it correct?

Yes.  But also double check all the relevant tests in case there's some state that the tests were dependent on.
